### PR TITLE
Refactored telemetry interface

### PR DESCRIPTION
**Pull request by ShreyasRmsft**

### CODE REVIEW

File: Agent.Plugins.Log.TestResultParser.Contracts/Telemetry/ITelemetryDataCollector.cs [hunk 1]
```diff
@@ -8,8 +8,28 @@ namespace Agent.Plugins.Log.TestResultParser.Contracts
...
+        /// Cumulates data throught the life cycle of the data collector until PublishCumulativeTelemetryAsync is called
+        /// This is used when the same data point is hit multiple times in an execution flow and firing an event each time
+        /// is a costly operations. This helps is agrregating the data that would otherwise have been done later and the 
+        /// inidividual data points themselves are of no interest.
+        /// </summary>
+        /// <param name="eventArea">Sub area of the data point</param>
+        /// <param name="eventName">Name of the data point</param>
+        /// <param name="value">Data point value to be stored</param>
+        /// <param name="aggregate">When to override or perform smart aggregation when the data point already exists</param>
         void AddToCumulativeTelemetry(string eventArea, string eventName, object value, bool aggregate = false);
```
Comment (cltshivash) [inline Agent.Plugins.Log.TestResultParser.Contracts/Telemetry/ITelemetryDataCollector.cs]: can be async as well..

Comment (ShreyasRmsft) [reply Agent.Plugins.Log.TestResultParser.Contracts/Telemetry/ITelemetryDataCollector.cs]: hmm okay. but who will wait on it?

Comment (ShreyasRmsft) [reply Agent.Plugins.Log.TestResultParser.Contracts/Telemetry/ITelemetryDataCollector.cs]: plus its a very light operation

Comment (nigurr) [inline Agent.Plugins.Log.TestResultParser.Contracts/Telemetry/ITelemetryDataCollector.cs]: if we are removing below, should we remove eventArea here?
EventArea property in interface?

Comment (ShreyasRmsft) [reply Agent.Plugins.Log.TestResultParser.Contracts/Telemetry/ITelemetryDataCollector.cs]: this is not same as area. Will rename to subarea or something

Comment (nigurr) [reply Agent.Plugins.Log.TestResultParser.Contracts/Telemetry/ITelemetryDataCollector.cs]: What's sub area? How each one of them co-relate to CustomerIntelligence Service values? Should we document how they map here

Comment (ShreyasRmsft) [reply Agent.Plugins.Log.TestResultParser.Contracts/Telemetry/ITelemetryDataCollector.cs]: Sub area and event name are concatenated to form the property name

example:

this._telemetry.AddToCumulativeTelemetry(JasmineTelemetryConstants.EventArea, "RegexTimeout", blah)

will have the property name 

JasmineTestResultParser:RegexTimeout

I can delegate this job to the folks who are calling this but this is more readable.

Comment (ShreyasRmsft) [reply Agent.Plugins.Log.TestResultParser.Contracts/Telemetry/ITelemetryDataCollector.cs]: yes let me document this here

**Pull request merged**